---
title: "Class 13: Genome Informatics (pt 1)"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Sample genotypes in the MXL 1000 Genome project data

Here we focus on the Mexican Ancestry in Los Angele, California (MXL) population.

Q5: What proportion of the Mexican Ancestry in Los Angeles sample population (MXL) are homozygous for the asthma associated SNP (G|G)?
```{r}
# Read CSV from ENSEMBLE
mxl <- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

How many of each genotype are there?
```{r}
table(mxl$Genotype..forward.strand.)
```

Proportion or percent of total for each genotype
```{r}
(table(mxl$Genotype..forward.strand.) / nrow(mxl)) * 100
```

## Quality Scores in FASTQ files

The fourth line of a FASTQ sequence format file encodes the quality score that tells us how good the sequence at a given position is. (i.e. how likely it is to be correct based on the instrument)

```{r}
library(seqinr)
library(gtools)
phred <- asc( s2c("DDDDCDEDCDDDDBBDDDCC@") ) - 33
phred
## D D D D C D E D C D D D D B B D D D C C @
## 35 35 35 35 34 35 36 35 34 35 35 35 35 33 33 35 35 35 34 34 31
prob <- 10**(-phred/10)
```


Section 4
```{r}
rs0867378 <- read.table("https://bioboot.github.io/bimm143_S18/class-material/rs8067378_ENSG00000172057.6.txt", row.names=1)

head(rs0867378)
```

```{r}
summary(rs0867378)
```

```{r}
inds <- rs0867378$geno =="G/G"
summary(rs0867378[inds, "exp"])
```

```{r}
inds <- rs0867378$geno =="A/G"
summary(rs0867378[inds, "exp"])
```

```{r}
boxplot(exp ~ geno , data=rs0867378)
```











