---
title: "Class 17"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Set up Cytoscape and R connection

We will use **RCy3** and **igraph** packages in this class.
The first is from bioconductor and the second is from CRAN

```{r}
library(RCy3)
library(igraph)
library(RColorBrewer)
```

```{r}
cytoscapePing()
```

```{r}
cytoscapeVersionInfo()
```

```{r}
g <- makeSimpleIgraph()
createNetworkFromIgraph(g,"myGraph")
```

```{r}
setVisualStyle("Marquee")
```


```{r}
fig <- exportImage(filename="demo", type="png", height=350)
knitr::include_graphics("./demo.png")
```


```{r}
g
```

```{r}
plot(g)
```


## Read our Metagenomics data

```{r}
prok_vir_cor <- read.delim("virus_prok_cor_abundant.tsv", stringsAsFactors = FALSE)

head(prok_vir_cor)
```


Use the **igraph.data.frame()** to make a network graph
```{r}
g <- graph.data.frame(prok_vir_cor, directed = FALSE)
```

```{r}
plot(g)
```


```{r}
g
```

Turn the node vertex labels off in this plot

```{r}
plot(g, vertex.label=NA)
```

Make the vertex smaller

```{r}
plot(g, vertex.size=3, vertex.label=NA)
```


```{r}
V(g)
```


```{r}
E(g)
```


```{r}
cb <- cluster_edge_betweenness(g)
```

```{r}
cb
```

```{r}
plot(cb, y=g, vertex.label=NA,  vertex.size=3)
```

```{r}
head( membership(cb) )
```

```{r}
d <- degree(g)
hist(d, breaks=30, col="lightblue", main ="Node Degree Distribution")
```

```{r}
plot( degree_distribution(g), type="h" )
```

## Read taxonomic classification for network annotation

```{r}
phage_id_affiliation <- read.delim("phage_ids_with_affiliation.tsv")
head(phage_id_affiliation)
```

```{r}
bac_id_affi <- read.delim("prok_tax_from_silva.tsv", stringsAsFactors = FALSE)
head(bac_id_affi)
```














